yield_plugin/src/yield_plugin.cpp

```diff
@@ -516,17 +519,22 @@ namespace yield_plugin
         {
           RCLCPP_WARN_STREAM(nh_->get_logger(), "Collision detected at timestamp " << std::to_string(p2a_t) << ", x: " << x1 << ", y: " << y1 << ", within distance: " << distance);
           return rclcpp::Time(p2a.header.stamp);
-        }      
+        }
       }
     }
 
     // No collision detected
     return std::nullopt;
   }
 
-  carma_planning_msgs::msg::TrajectoryPlan YieldPlugin::update_traj_for_object(const carma_planning_msgs::msg::TrajectoryPlan& original_tp, const std::vector<carma_perception_msgs::msg::ExternalObject>& external_objects, double initial_velocity) 
-  {    
-    if (original_tp.trajectory_points.empty())
+  carma_planning_msgs::msg::TrajectoryPlan YieldPlugin::update_traj_for_object(const carma_planning_msgs::msg::TrajectoryPlan& original_tp, const std::vector<carma_perception_msgs::msg::ExternalObject>& external_objects, double initial_velocity)
+  {
+    if (original_tp.trajectory_points.size() < 2)
+    {
+      RCLCPP_WARN(nh_->get_logger(), "Yield plugin received less than 2 points in update_traj_for_object, returning unchanged...");
+      return original_tp;
+    }
+    else if (original_tp.trajectory_points.empty())
     {
       RCLCPP_ERROR(nh_->get_logger(), "Yield plugin received empty trajectory plan in update_traj_for_object");
       throw std::invalid_argument("Yield plugin received empty trajectory plan in update_traj_for_object");
```

Comment:
This does not need to be an `if...else` because the `if`'s body will exit the function. This can be simplified to two independent `if` statements.